CMakeLists.txt

```diff
@@ -3,19 +3,46 @@ cmake_minimum_required(VERSION 3.14)
 # set the project name and version
 project(QSsh VERSION 1.0 LANGUAGES CXX)
 
+# use GNU installation dirs
+include(GNUInstallDirs)
+
+# use package configuration helpers
+include(CMakePackageConfigHelpers)
+
+# cached variables
+set(QSSH_BUILD_EXAMPLES True CACHE BOOL "Build the QSsh examples.")
+set(BUILD_SHARED_LIBS False CACHE BOOL "Build QSsh using shared libraries.")
+set(BOTAN_INCLUDE_DIR "" CACHE PATH "Path of the Botan include directory. Set this if user specified BOTAN_LIBRARY location is used.")
```

Comment:
Why not to use find package?

Reply:
Last time I checked Botan-2 does not provide .cmake files for finding package. So if I build Botan-2 manually I need a way to point QSsh to the version I have built.

Reply:
You can provide your own Findbotan.cmake (example https://stackoverflow.com/a/72323666). botan is a thirdparty library so it should not be in main cmakelists

Reply:
Great idea! Could be done in a separate MR though, this one is already changing a bit.